quic/api/Observer.h

```diff
@@ -72,6 +72,21 @@ class InstrumentationObserver {
         lastAckedPacketInfo;
   };
 
+  struct ObserverSpuriousLossEvent {
+    explicit ObserverSpuriousLossEvent(TimePoint rcvTimeIn)
+        : rcvTime(rcvTimeIn) {}
+    TimePoint rcvTime;
+
+    bool hasPackets() {
+      return spuriousPackets.size() > 0;
+    }
+
+    void addSpuriousPacket(quic::OutstandingPacket pkt) {
```

Comment:
Do we have information about the spurious loss that we can also add, like why the packet was originally marked lost?

Reply:
i can save the reason in the packet when loss is detected. lostByTimeout and lostByReorder in LostPacket would be a convenience as the reason is already inside the OutstandingPacket.

Reply:
Use `const quic::OutstandingPacket& pkt`?

Reply:
Use `const quic::OutstandingPacket& pkt`?